jfix-stdlib-concurrency/src/test/kotlin/ru/fix/stdlib/concurrency/events/EventReducerTest.kt

```diff
@@ -0,0 +1,154 @@
+package ru.fix.stdlib.concurrency.events
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import ru.fix.aggregating.profiler.NoopProfiler
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+
+private const val INVOCATION_TIMEOUT_MS = 500L
+
+internal class EventReducerTest {
+
+    @Test
+    fun `WHEN handle single time THEN handler invoked one time`() {
+        val invokes = ArrayBlockingQueue<Any>(1)
```

Comment:
You set size ArrayBlockingQueue<Any>(1) to one. Why? 
What is a purpose of blocking handler invocation?
Are you testing that they should work in different threads in some kind of concurrency issue here? 
If all you need is to check how many times handler was invoked then use LinkedBlockingQueue.
And simply check size and content at the end of the test.

Reply:
ArrayBlockingQueue removed due to removing internal thread